runelite-client/src/main/java/net/runelite/client/plugins/agility/AgilityPlugin.java

```diff
@@ -421,4 +426,41 @@ private void onTileObject(Tile tile, TileObject oldObject, TileObject newObject)
 			}
 		}
 	}
+
+	@Subscribe
+	public void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (!config.showShortcutLevel())
+		{
+			return;
+		}
+
+		//Guarding against non-first option because agility shortcuts are always that type of event.
+		if (event.getType() != MenuAction.GAME_OBJECT_FIRST_OPTION.getId())
+		{
+			return;
+		}
+
+		final int entryId = event.getIdentifier();
+		MenuEntry[] menuEntries = client.getMenuEntries();
+
+		for (Obstacle nearbyObstacle : getObstacles().values())
+		{
+			AgilityShortcut shortcut = nearbyObstacle.getShortcut();
+			if (Arrays.stream(shortcut.getObstacleIds()).anyMatch(i -> i == entryId))
```

Comment:
Apparently getObstacleIds can return null resulting in an NPE here, it's definitely worth a little refactor of AgilityShortcut to use ImmutableList or ImmutableSet instead of an array to take care of multiple issues.